v2/internal/util/metrictree/directmatch_test.go

```diff
@@ -0,0 +1,136 @@
+// +build small
+
+/*
+ Copyright (c) 2020 SolarWinds Worldwide, LLC
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package metrictree
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+type testCase struct {
+	ns          string
+	filter      string
+	shouldMatch bool
+	expectError bool
+}
+
+var testCases = []testCase{
+	{
+		ns:          "/kubernetes/[pod=io1453]/cpu",
+		filter:      "/kubernetes/[pod=io1453]/cpu",
+		shouldMatch: true,
+	},
+	{
+		ns:          "/kubernetes/[pod=io1453]/cpu",
+		filter:      "/kubernetes/[pod=io1453]",
+		shouldMatch: true,
+	},
+	{
+		ns:          "/kubernetes/[pod=io1453]/cpu",
+		filter:      "/kubernetes/[pod]",
+		shouldMatch: true,
+	},
+	{
+		ns:          "/kubernetes/[pod=io1453]/cpu",
+		filter:      "/kubernetes/io1453",
+		shouldMatch: true,
+	},
+	{
+		ns:          "/kubernetes/[pod=io1453]/cpu",
+		filter:      "/kubernetes",
+		shouldMatch: true,
+	},
+	{
+		ns:          "/kubernetes/[pod=io1453]/cpu",
+		filter:      "/",
+		shouldMatch: true,
+	},
+	{
+		ns:          "|kubernetes|[pod=io1453]|cpu",
+		filter:      "|kubernetes|[pod={io[0-9]{4}}]|cpu",
+		shouldMatch: true,
+	},
+	{
+		ns:          "/kubernetes//cpu",
+		filter:      "/kubernetes/[pod={io[0-9]{4}}]/cpu",
+		expectError: true,
+	},
+	{
+		ns:          "|log-files|[file=/tmp/sample.log]|string_line",
+		filter:      "|log-files|[file=/tmp/sample.log]",
+		shouldMatch: true,
+	},
+	{
+		ns:          "|log-files|[file=/tmp/sample1.log]|string_line",
+		filter:      "|log-files|[file=/tmp/sample.log]|string_line",
+		shouldMatch: false,
+	},
+	{
+		ns:          "/syslog/[ip_address=127.0.0.1]/[hostname=localhost]/string_line",
+		filter:      "/syslog/*/[hostname=localhost]/string_line",
```

Comment:
please add a test case (with the same `ns`):

```suggestion
		filter:      "/syslog/**/string_line",
```

Reply:
done

Reply:
** cannot be in the middle, it is for last element

Reply:
ah, ok